sigilrecorder.lic

```diff
@@ -0,0 +1,205 @@
+		require 'yaml'
+		@data = YAML.load(File.read("./scripts/data/base-sigils.yaml"))
+		@city_list = ['shard', 'crossing', 'riverhaven', 'other']
+		@sigil_list = ['abolition','congruence','induction','permutation','rarefaction', 'antipode','ascension','clarification','decay','evolution','integration','metamorphosis','nurture','paradox','unity']
+		@settings = get_settings
+		get_region
+		get_season #make sure we're recording sigils to the correct season
+		region_room_list #generate the walk_list array
+		generate_nogo_list
+		
+		Flags.add('danger-room', /.*begins to advance on you.*|You weave back and forth in a dance-like motion|You must be closer to use tactical abilities on your opponent/)
+		@walk_list.each do |roomid|
+			@data['nogo'].include?(roomid) ? next : (travel_result = walk_to(roomid, false))
+			respond("Travel Debug: #{travel_result} | Room Target: #{roomid}")
+			if Room.current.id == nil
+			  respond("Adding #{roomid} to nogo due to lack of visible room number in LOOK.")
+			  add_to_nogo(roomid)
+			  unstuck_me(roomid)
+			  next
+			end
+			if travel_result == true
+				@current_room = Room.current.id
+				DRCA.do_buffs(@settings, 'outdoors')
+				if check_room_data < 2
+				  @room_sigils = []
+				  get_season
+				  if find_sigil?
+				    next #skip this room because something went wrong or it is dangerous
+				  end
+				  respond("This room contains the following sigils #{@room_sigils}")
+				else
+				  respond("Sigils in this room already known")
+				end #if
+			else
+				respond('Unable to reach target room. Adding to no-go list and moving to next room.')
+				add_to_nogo(Room.current.id)
+				next
+			end #if
+			
+			write_to_file
+		end #do
+	end #initialize
+		
+		
+	def get_season
+		/^It is currently (\w+) and it is.*$/ =~ bput('time', /^It is currently.*/)
+		@season = $1
+	end #get_season
+	
+	
+	def get_region
+		case Room.current.image.downcase
+		when /ilithi|shard/
+		  @region = 'shard'
+		when /zoluren|crossing/
+		  @region = 'crossing'
+		when /therengia|riverhaven/
+		  @region = 'riverhaven'
+		else
+		  @region = 'other'
+		end #case
+		respond("Region detected: #{@region.capitalize}")
+	end
+	
+	
+	def check_room_data
+		room_array = []
+		room_matches = 0
+		@city_list.each do |city|
+		  @sigil_list.each do |sigil|
+		    @data.dig(city, sigil, @season).each do |number|
+			  if number != nil
+			    room_array.push(number)
+			  end
+			next
+			end
+		  next
+		  end
+		next
+		end
+		
+		room_array.compact
+		#respond("Full Room List: #{room_array}")
+		room_matches = room_array.count(@current_room)
+
+		respond("Current room (#{@current_room}) matches in data = #{room_matches}")
+		
+		return room_matches
+		
+	end #check_room_data
+	
+	def unstuck_me(target_room)
+		exit_list = Map[target_room].wayto
+		exit_list.each do |rnum, dir|
+		  Flags.add('room-detected', /Room Number: \d+/)
+		  fput("peer #{dir}")
+		  pause
+		  if Flags['room-detected']
+		    fput('retreat')
```

Comment:
We have a nice common for retreating DRC.retreat. Can use that here instead of the fput? (search and replace all)